drivers/flash/flash_esp32.c

```diff
@@ -210,6 +227,83 @@ static int flash_esp32_erase(const struct device *dev, off_t start, size_t len)
 	return 0;
 }
 
+#ifdef CONFIG_ESP_FLASH_ASYNC
+static void flash_work_handler(struct k_work *work)
+{
+	struct flash_esp32_dev_data *data = CONTAINER_OF(work, struct flash_esp32_dev_data, work);
+
+	if (data->type == 0) {
+		data->ret = flash_esp32_read(data->dev, data->addr, data->buf, data->len);
+	} else if (data->type == 1) {
+		data->ret = flash_esp32_write(data->dev, data->addr, data->buf, data->len);
+	} else if (data->type == 2) {
+		data->ret = flash_esp32_erase(data->dev, data->addr, data->len);
+	} else {
+		data->ret = -1;
+	}
```

Comment:
Is it possible to create defines or a enum for the type field? I know this will not change often, but it will keep this section better for reading.

Reply:
I was considering this aw but didn't make it for the sake of memory footprint. Maybe introducing a new `CONFIG_ESP_FLASH_WORKQUEUE_DEDICATED` orso do it. Any thoughts @uLipe ?

Reply:
Yes, I was thinking on something similar, if you have enough memory go for a dedicated workqueue, otherwise use the system workqueue.